src/ray/raylet/lineage_cache.cc

```diff
@@ -296,10 +294,14 @@ bool LineageCache::SubscribeTask(const TaskID &task_id) {
   auto inserted = subscribed_tasks_.insert(task_id);
   bool unsubscribed = inserted.second;
   if (unsubscribed) {
-    // Request notifications for the task if we haven't already requested
-    // notifications for it.
-    RAY_CHECK_OK(task_pubsub_.RequestNotifications(JobID::Nil(), task_id, client_id_,
-                                                   /*done*/ nullptr));
+    auto subscribe = [this](const TaskID &task_id, const TaskTableData) {
+      HandleEntryCommitted(task_id);
+    };
+    // Subscribe to the task.
+    // TODO(micafan) If we commit task and subscribe task at the same time,
+    // we might not receive the commited message.
```

Comment:
By "at the same time", do you mean in different threads?

Reply:
Actually I don't remember why I added such a comment here.
In fact, we will receive the message regardless  the order of `Subscribe` and `Commit` . 
I will remove the comment.